### PR TITLE
Functional test kernel - fission/fusion

### PR DESCRIPTION
**Pull request by artv3**
Functional test for Kernel fission/fusion (includes basic + conditional)

### CODE REVIEW

**Comment by artv3** [comment]
Yes! It passes! One note, I wasn't sure how many policies to use for testing, I just used a few. I can certainly add more if needed.

**Comment by rhornung67** [comment]
@artv3 We are using a reduced set of exec policies, index types, etc. for the default build case for most tests. We have a CMake variable https://github.com/LLNL/RAJA/blob/develop/CMakeLists.txt#L77 to enable a larger set of tests. Go ahead and add more policies that you think would be good to test and guard them with the "test exhaustive" macro variable.

**Comment by rhornung67** [comment]
@ajkunen @mdavis36 @rchen20 I think this is ready to merge. Please review. Thank you.

**Review by rhornung67 — DISMISSED** [review]

**Review by ajkunen — APPROVED** [review]

File: test/functional/kernel/basic-fission-fusion-loop/test-kernel-basic-fission-fusion-loop.cpp.in [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+using @BACKEND@KernelBasicFisionFussionLoopTypes =
+    Test<camp::cartesian_product<StrongIdxTypeList,
+                                 @BACKEND@ResourceList,
+                                 @BACKEND@KernelExecPols>>::Types;
+
+//
+// Instantiate parameterized tests
+//
+INSTANTIATE_TYPED_TEST_SUITE_P(@BACKEND@,
+                               KernelBasicFisionFussionLoopTest,
```
Comment (rchen20) [inline test/functional/kernel/basic-fission-fusion-loop/test-kernel-basic-fission-fusion-loop.cpp.in:174]: Misspelling of `FisionFussion` here and throughout. I'll fix these.

Comment (rhornung67) [reply test/functional/kernel/basic-fission-fusion-loop/test-kernel-basic-fission-fusion-loop.cpp.in:174]: Thanks. I missed that. Sorry.

**Review by rchen20 — APPROVED** [review]

**Review by rhornung67 — APPROVED** [review]

**Pull request merged**